s3tests_boto3/functional/test_s3.py

```diff
@@ -8571,6 +8571,52 @@ def test_encryption_sse_c_multipart_upload():
     _check_content_using_range_enc(client, bucket_name, key, data, 1000000, enc_headers=enc_headers)
     _check_content_using_range_enc(client, bucket_name, key, data, 10000000, enc_headers=enc_headers)
 
+@attr(resource='object')
+@attr(method='put')
+@attr(operation='complete multi-part upload with unaligned part size')
+@attr(assertion='successful')
+@attr('encryption')
+def test_encryption_sse_c_multipart_upload():
+    bucket_name = get_new_bucket()
+    client = get_client()
+    key = "multipart_enc"
+    content_type = 'text/plain'
+    objlen = 30 * 1024 * 1024
+    partlen = 1 + 5 * 1024 * 1024
```

Comment:
one more case to add would be to set partlen as just 5MiB and objlen as say something with an  offby1 , say 30MiB+ 10 which should catch the case of the last unaligned part.

Reply:
i now have the original multipart tests using objlen=30M+1 partlen=5M, and the new unaligned tests using objlen=30M partlen=5M+1